### ISSUE TITLE
Feature request: disable-comments that expect errors

### ISSUE THREAD

**Issue by mightyiam** [eslint/eslint#13672]
**The version of ESLint you are using.**

v7.8.1

**The problem you want to solve.**

Let's say that I used one of the "disable" comments, like `// eslint-disable-next-line`, for example, for the purpose of ignoring a false-positive. And let's say that that specific rule was updated/fixed in a more recent version and it no longer produces this false-positive. And let's say that I have upgraded to that version so that now the disable-comment is no longer necessary. The problem is that I could neglect removing that disable-comment. Not a big deal — especially if specifying a rule name, like

```js
// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

But, still, I would like a mechanism that let's me know that this comment is no longer necessary.

**Your take on the correct solution to problem.**

Perhaps a new kind of comment, like:

```js
// eslint-expect-next-line @typescript-eslint/no-unused-vars
```

This new kind of comment expects some rule or a specific rule(s) to error. If no such error, then the comment itself will error. I get the inspiration for this solution from [TypeScript's `// @ts-expect-error` comments](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments).

Another possible solution is a new configuration option that changes the behavior of all current disable-comments to _expect_ errors.

**Are you willing to submit a pull request to implement this change?**

Possibly.

**Comment by anikethsaha** [comment]
Thanks for the issue. 

is it similar to  [--report-unused-disable-directives](https://eslint.org/docs/user-guide/command-line-interface#-report-unused-disable-directives-1) flag of the CLI ?

**Comment by mightyiam** [comment]
> Thanks for the issue. 
> 
> is it similar to  [--report-unused-disable-directives](https://eslint.org/docs/user-guide/command-line-interface#-report-unused-disable-directives-1) flag of the CLI ? cause if there are some false cases i.e false positive and that is fixed in some version, so that means that code block wont be reported by that rule. 

Oh, such a feature exists, I see. Well, then I guess this issue may be closed. Though it is not clear to me why this is only a command line flag and not a configuration option.

**Comment by mdjermanovic** [comment]
There's also configuration option [reportUnusedDisableDirectives](https://eslint.org/docs/user-guide/configuring#report-unused-eslint-disable-comments).

**Comment by mightyiam** [comment]
Thank you!
